state_machine/src/lib.rs

```diff
@@ -114,42 +108,44 @@ pub struct OverlayedChanges {
 	committed: MemoryState,
 }
 
+const EMPTY_BYTES: [u8; 0] = [];
+
 impl OverlayedChanges {
-	fn code(&self, address: &Address) -> Option<&[u8]> {
-		self.prospective.code(address)
-			.or_else(|| self.committed.code(address))
-			.and_then(|v| if v.is_empty() { None } else { Some(v) })
+	fn code(&self) -> &[u8] {
+		self.prospective.code()
+			.or_else(|| self.committed.code())
+			.unwrap_or_else(|| &EMPTY_BYTES)
```

Comment:
Does `&[]` not compile? what about `&[0u8; 0]`?

Reply:
reference couldn't be taken out of unwrap's closure.

Reply:
just `unwrap_or` then

Reply:
It seems like it compiles anyway: https://play.rust-lang.org/?gist=4db42228e4f949306f64ede8b315e7a4&version=stable